### PR TITLE
Conversion to ES6

### PR DESCRIPTION
**Pull request by Edmondton**
@longlho @DekusDenial @redonkulus

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "stage": 0
```
Comment (redonkulus) [inline .babelrc]: be careful with stage 0 as they are just proposals still.

Comment (Edmondton) [inline .babelrc]: should I remove it entirely ? @redonkulus

Comment (redonkulus) [inline .babelrc]: you can use it, but know its unstable.

**Comment by longlho** [comment]
hmm the `Map.jsx` is still `jsx`, not `js` thou.

**Comment by yahoocla** [comment]
CLA is valid!

File: index.jsx [hunk 2]
```diff
@@ -2,92 +2,88 @@
...
-            addressLine1: '701 First Avenue',
-            city: 'Sunnyvale',
-            state: 'CA',
-            country: 'United States',
-            zoom: 15,
-            height: 270,
-            width: 580
-        })}
+        <Map
+            provider={'yahoo'}
```
Comment (longlho) [inline index.jsx]: can u do `provider="yahoo"` like normal markups?

File: src/components/Map.jsx [hunk 3]
```diff
@@ -5,22 +5,14 @@
...
+import React from 'react';
+import MapLocationFactory from '../lib/mapLocationFactory';
 
 // Factory
-var factory = require('../lib/mapLocationFactory');
-
-module.exports = React.createClass({
-    displayName: 'Map',
-    mixins: [PureRenderMixin],
+var factory = new MapLocationFactory();
```
Comment (longlho) [inline src/components/Map.jsx]: `const factory`?

File: src/components/Map.jsx [hunk 4]
```diff
@@ -33,23 +25,17 @@ module.exports = React.createClass({
         height: React.PropTypes.number.isRequired,
         width: React.PropTypes.number.isRequired,
         zoom: React.PropTypes.number
-    },
+    };
```
Comment (longlho) [inline src/components/Map.jsx]: `;` is not necessary I think

File: src/lib/mapLocationFactory.js [hunk 5]
```diff
@@ -5,9 +5,13 @@
 
 'use strict';
 
-var yahooService    = require('./yahooMap');
-var googleService   = require('./googleMap');
-var bingService     = require('./bingMap');
+import BingMapService from './bingMap';
```
Comment (longlho) [inline src/lib/mapLocationFactory.js]: does `import './bingMap' as BingMapService` work?

File: .npmignore [hunk 6]
```diff
@@ -4,4 +4,6 @@
 /tests/
 lcov-*
 xunit*
-components/
\ No newline at end of file
+components/
+src/
```
Comment (longlho) [inline .npmignore]: can we keep this in the package too? Just in case someone is full ES6 wants to use the src version

**Comment by longlho** [comment]
:+1:

**Comment by Edmondton** [comment]
dear @longlho thank you for your review.

**Pull request merged**